src/Logic.hs

```diff
@@ -817,9 +829,14 @@ tryIntegratePullRequest pr state =
       Right (Sha sha) -> do
         -- If it succeeded, set the build to pending,
         -- as pushing should have triggered a build.
+        let Pr.MandatoryChecks check = Pr.mandatoryChecks state
+            integrationStatus = if Set.null check
+              then Pr.AnyCheck BuildPending
+              else Pr.SpecificChecks Map.empty
```

Comment:
Part of the issue seems to be this initialization. What I had in mind is that we initialize this map with all the mandatory checks being pending.
That way, we, at all later times, know exactly which checks are already done, and which we're still waiting for.